AbletonMCP_Remote_Script/__init__.py

```
@@ -799,8 +829,340 @@ def _find_browser_item_by_uri(self, browser_or_item, uri, max_depth=10, current_
+                "sample_name": item.name if hasattr(item, 'name') else "Unknown",
+                "track_index": track_index
+            }
+        except Exception as e:
+            self.log_message("Error loading sample: " + str(e))
+            raise
+
+    def _create_clip_automation(self, track_index, clip_index, parameter_name, automation_points):
+        """Create automation envelope in a clip"""
+        try:
+            if track_index < 0 or track_index >= len(self._song.tracks):
+                raise IndexError("Track index out of range")
```

Ruff (0.14.14)

[warning] Abstract `raise` to an inner function

(TRY301)